### DISCUSSION TITLE
What am I missing here?

### DISCUSSION THREAD

**Discussion by stdedos** [python-gitlab/python-gitlab#2917] [Q&A]
I am trying to probe the https://docs.gitlab.com/ee/api/merge_requests.html#get-single-mr API, and I thought I'd use `python-gitlab` for this.

After some trying, I got to

```bash
gitlab -v --gitlab inst-gitlab project-merge-request get --project-id 15 --iid 15899
```

I expected "some kind of output". However, I only received `iid: 15899` - which I kinda new already 😕 

I needed to add `-v` to see "some kind of output". Then, I noticed `-o json`, and I got something of what I wanted.

... However, why isn't that the default? I don't want to start a "default output format" religious war - but **some kind of output**, that's not what I needed to spoon-feed in the first place

What am I missing here?

**Comment by nejch** [comment]
@stdedos in the legacy output (the default), the default behavior is to print out the ID and what we call the `_repr_attr` which is the human-readable attribute. It just so happens that merge request object on our end doesn't have a repr attribute defined (https://github.com/python-gitlab/python-gitlab/blob/main/gitlab/v4/objects/merge_requests.py#L155), so all you see is an ID. Ideally this would be the title, for example.

I'm not sure what the decision behind the short vs. verbose behavior was as it's before my time. YAML would probably make more sense and just print it all out, but that pulls in another (currently optional) dependency for the CLI.

**Reply by stdedos** [reply]
... or JSON and avoid the extra dependency 🙃 

Could we extend the config file with a global (and per-server?) option `format = legacy | json | yaml`?

**Reply by nejch** [reply]
I guess JSON is more for programmatic use so might not be the nicest output by default :sweat_smile:  Yes, a global config option might be best actually.

**Reply by stdedos** [reply]
> Yes, a global config option might be best actually.

I just "don't want to remember this" anymore 😅 

Maybe I'm a minority, but I'm using the CLI solely as the "quick" version of `curl...` (trying to figure out how to build a query) 

Legacy format (if extended) may make sense. Idk the amount of "ready formats" vs "not-ready" (like my command's case). 

If not-ready is a majority, it's better IMHO to switch to YAML (if you are fine for the dependency). 

Honestly, I was "mislead" thinking that I can pass posargs to the command and get the keys I wanted 😅
